src/nvim/path.c

```diff
@@ -481,6 +481,24 @@ static size_t path_expand(garray_T *gap, const char_u *path, int flags)
   return do_path_expand(gap, path, 0, flags, false);
 }
 
+static const char *os_scandir_next_with_dots(Directory *dir)
```

Comment:
`scandir_next_with_dots` since it's a private function, not in "os" module.

This might be paranoia, but in case it is not called >=3 times, it will be left in a bad state. So there should be a way to "initialize" the internal state. I suggest if `dir == NULL` then `count` should be initialized to 0.